File: MiniDOMOffice/Scansioni/ScansioniCursor.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Office;

namespace minidom
{
    public partial class Office
    {


        /// <summary>
        /// Cursore di <see cref="Scansione"/>
        /// </summary>
        /// <remarks></remarks>
        [Serializable]
        public class ScansioneCursor 
            : minidom.Databases.DBObjectCursorPO<Scansione>
        {
            private DBCursorStringField m_NomeDispositivo = new DBCursorStringField("NomeDispositivo");
            private DBCursorStringField m_NomeDocumento = new DBCursorStringField("NomeDocumento");
            private DBCursorStringField m_MetodoRicezione = new DBCursorStringField("MetodoRicezione");
            private DBCursorStringField m_ParametriScansione = new DBCursorStringField("ParametriScansione");
            private DBCursorField<DateTime> m_DataInvio = new DBCursorField<DateTime>("DataInvio");
            private DBCursorField<DateTime> m_DataRicezione = new DBCursorField<DateTime>("DataRicezione");
            private DBCursorField<DateTime> m_DataElaborazione = new DBCursorField<DateTime>("DataElaborazione");
            private DBCursorField<int> m_IDCliente = new DBCursorField<int>("IDCliente");
            private DBCursorStringField m_NomeCliente = new DBCursorStringField("NomeCliente");
            private DBCursorField<int> m_IDAttachment = new DBCursorField<int>("IDAttachment");
            private DBCursorField<int> m_Flags = new DBCursorField<int>("Flags");
            private DBCursorField<int> m_IDInviataDa = new DBCursorField<int>("IDInviataDa");
            private DBCursorStringField m_NomeInviataDa = new DBCursorStringField("NomeInviataDa");
            private DBCursorField<int> m_IDInviataA = new DBCursorField<int>("IDInviataA");
            private DBCursorStringField m_NomeInviataA = new DBCursorStringField("NomeInviataA");
            private DBCursorField<int> m_IDElaborataDa = new DBCursorField<int>("IDElaborataDa");
            private DBCursorStringField m_NomeElaborataDa = new DBCursorStringField("NomeElaborataDa");

            /// <summary>
            /// Costruttore
            /// </summary>
            public ScansioneCursor()
            {
            }

            /// <summary>
            /// NomeDispositivo
            /// </summary>
            public DBCursorStringField NomeDispositivo
            {
                get
                {
                    return m_NomeDispositivo;
                }
            }

            /// <summary>
            /// NomeDocumento
            /// </summary>
            public DBCursorStringField NomeDocumento
            {
                get
                {
                    return m_NomeDocumento;
                }
            }

            /// <summary>
            /// MetodoRicezione
            /// </summary>
            public DBCursorStringField MetodoRicezione
            {
                get
                {
                    return m_MetodoRicezione;
                }
            }

            /// <summary>
            /// ParametriScansione
            /// </summary>
            public DBCursorStringField ParametriScansione
            {
                get
                {
                    return m_ParametriScansione;
                }
            }

            /// <summary>
            /// DataInvio
            /// </summary>
            public DBCursorField<DateTime> DataInvio
            {
                get
                {
                    return m_DataInvio;
                }
            }

            /// <summary>
            /// DataRicezione
            /// </summary>
            public DBCursorField<DateTime> DataRicezione
            {
                get
                {
                    return m_DataRicezione;
                }
            }

            /// <summary>
            /// DataElaborazione
            /// </summary>
            public DBCursorField<DateTime> DataElaborazione
            {
                get
                {
                    return m_DataElaborazione;
                }
            }

            /// <summary>
            /// IDCliente
            /// </summary>
            public DBCursorField<int> IDCliente
            {
                get
                {
                    return m_IDCliente;
                }
            }

            /// <summary>
            /// NomeCliente
            /// </summary>
            public DBCursorStringField NomeCliente
            {
                get
                {
                    return m_NomeCliente;
                }
            }

            /// <summary>
            /// IDAttachment
            /// </summary>
            public DBCursorField<int> IDAttachment
            {
                get
                {
                    return m_IDAttachment;
                }
            }

            /// <summary>
            /// Flags
            /// </summary>
            public DBCursorField<int> Flags
            {
                get
                {
                    return m_Flags;
                }
            }

            /// <summary>
            /// IDInviataDa
            /// </summary>
            public DBCursorField<int> IDInviataDa
            {
                get
                {
                    return m_IDInviataDa;
                }
            }

            /// <summary>
            /// NomeInviataDa
            /// </summary>
            public DBCursorStringField NomeInviataDa
            {
                get
                {
                    return m_NomeInviataDa;
                }
            }

            /// <summary>
            /// IDInviataA
            /// </summary>
            public DBCursorField<int> IDInviataA
            {
                get
                {
                    return m_IDInviataA;
                }
            }

            /// <summary>
            /// NomeInviataA
            /// </summary>
            public DBCursorStringField NomeInviataA
            {
                get
                {
                    return m_NomeInviataA;
                }
            }

            /// <summary>
            /// IDElaborataDa
            /// </summary>
            public DBCursorField<int> IDElaborataDa
            {
                get
                {
                    return m_IDElaborataDa;
                }
            }

            /// <summary>
            /// NomeElaborataDa
            /// </summary>
            public DBCursorStringField NomeElaborataDa
            {
                get
                {
                    return m_NomeElaborataDa;
                }
            }

            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Office.Scansioni;
            }
             
        }
    }
}

File: MiniDOMOffice/Scansioni/ScansioniCursor.vb
Imports minidom.Databases
Imports minidom.Sistema

Partial Class Office


    ''' <summary>
    ''' Cursore sulla tabella dei scansioni
    ''' </summary>
    ''' <remarks></remarks>
    Public Class ScansioneCursor
        Inherits DBObjectCursorPO(Of Scansione)

        Private m_NomeDispositivo As New CCursorFieldObj(Of String)("NomeDispositivo")
        Private m_NomeDocumento As New CCursorFieldObj(Of String)("NomeDocumento")
        Private m_MetodoRicezione As New CCursorFieldObj(Of String)("MetodoRicezione")
        Private m_ParametriScansione As New CCursorFieldObj(Of String)("ParametriScansione")
        Private m_DataInvio As New CCursorField(Of Date)("DataInvio")
        Private m_DataRicezione As New CCursorField(Of Date)("DataRicezione")
        Private m_DataElaborazione As New CCursorField(Of Date)("DataElaborazione")
        Private m_IDCliente As New CCursorField(Of Integer)("IDCliente")
        Private m_NomeCliente As New CCursorFieldObj(Of String)("NomeCliente")
        Private m_IDAttachment As New CCursorField(Of Integer)("IDAttachment")
        Private m_Flags As New CCursorField(Of Integer)("Flags")
        Private m_IDInviataDa As New CCursorField(Of Integer)("IDInviataDa")
        Private m_NomeInviataDa As New CCursorFieldObj(Of String)("NomeInviataDa")
        Private m_IDInviataA As New CCursorField(Of Integer)("IDInviataA")
        Private m_NomeInviataA As New CCursorFieldObj(Of String)("NomeInviataA")
        Private m_IDElaborataDa As New CCursorField(Of Integer)("IDElaborataDa")
        Private m_NomeElaborataDa As New CCursorFieldObj(Of String)("NomeElaborataDa")


        Public Sub New()
        End Sub

        Public ReadOnly Property NomeDispositivo As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeDispositivo
            End Get
        End Property

        Public ReadOnly Property NomeDocumento As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeDocumento
            End Get
        End Property

        Public ReadOnly Property MetodoRicezione As CCursorFieldObj(Of String)
            Get
                Return Me.m_MetodoRicezione
            End Get
        End Property

        Public ReadOnly Property ParametriScansione As CCursorFieldObj(Of String)
            Get
                Return Me.m_ParametriScansione
            End Get
        End Property

        Public ReadOnly Property DataInvio As CCursorField(Of Date)
            Get
                Return Me.m_DataInvio
            End Get
        End Property

        Public ReadOnly Property DataRicezione As CCursorField(Of Date)
            Get
                Return Me.m_DataRicezione
            End Get
        End Property

        Public ReadOnly Property DataElaborazione As CCursorField(Of Date)
            Get
                Return Me.m_DataElaborazione
            End Get
        End Property

        Public ReadOnly Property IDCliente As CCursorField(Of Integer)
            Get
                Return Me.m_IDCliente
            End Get
        End Property

        Public ReadOnly Property NomeCliente As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeCliente
            End Get
        End Property

        Public ReadOnly Property IDAttachment As CCursorField(Of Integer)
            Get
                Return Me.m_IDAttachment
            End Get
        End Property

        Public ReadOnly Property Flags As CCursorField(Of Integer)
            Get
                Return Me.m_Flags
            End Get
        End Property

        Public ReadOnly Property IDInviataDa As CCursorField(Of Integer)
            Get
                Return Me.m_IDInviataDa
            End Get
        End Property

        Public ReadOnly Property NomeInviataDa As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeInviataDa
            End Get
        End Property

        Public ReadOnly Property IDInviataA As CCursorField(Of Integer)
            Get
                Return Me.m_IDInviataA
            End Get
        End Property

        Public ReadOnly Property NomeInviataA As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeInviataA
            End Get
        End Property

        Public ReadOnly Property IDElaborataDa As CCursorField(Of Integer)
            Get
                Return Me.m_IDElaborataDa
            End Get
        End Property

        Public ReadOnly Property NomeElaborataDa As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeElaborataDa
            End Get
        End Property

        Protected Overrides Function GetModule() As CModule
            Return Office.Scansioni.Module
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_OfficeScansioni"
        End Function

        Protected Overrides Function GetConnection() As CDBConnection
            Return Office.Database
        End Function
    End Class



End Class